lib/keycloak.js

```diff
@@ -1237,7 +1223,9 @@ export default class Keycloak {
 
         const params = new URLSearchParams([
             ['client_id', /** @type {string} */ (this.clientId)],
-            ['redirect_uri', redirectUri],
+            // The endpoint URI MUST NOT include a fragment component.
+            // https://datatracker.ietf.org/doc/html/rfc6749#section-3.1.2
+            ['redirect_uri', stripHash(redirectUri)],
```

Comment:
Good catch and great addition! 🤓 
I think that is actually helping us resolving an issue with our Firewall.

Reply:
I see the point of the specs compliance, but it looks like this is backwards incompatible change if I understand correctly?

For example today when I open http://localhost:8080/admin/master/console/#/foo/clients and then I login as admin/admin, I am redirected directly to the page http://localhost:8080/admin/master/console/#/foo/clients and I see the list of clients of realm `foo` . But if I understand correctly, with this change, I would be instead redirected to the default admin console page http://localhost:8080/admin/master/console/ and the fragment context is lost. Or am I wrong here?

If fragment causes issues in some environments, then it might be perhaps configurable whether to automatically strip fragment or not?